ipq40xx: add support for Asus Lyra Mini (MAP-AC1300)

sbeach92 opened the pull request:
Hardware
CHIPSET: IPQ4018 + QCA8072
LAN: 2x 1G
FLASH: GD5F1GQ4X 128 MiB + MX25L165D 2 MiB
RAM: 256 MiB
IO: Reset + WPS button
LED: Single Bright RGB led
POWER: 5V 2A

Flash layout similar to other asus brand routers.

Mac adresses extarcted from same mtd adress.

Orginal -> Openwrt
ath0 2C:FD:A1:8D:80:B8 -> phy0-ap0 2C:FD:A1:8D:80:B8
ath1 2C:FD:A1:8D:80:BA -> phy1-ap0 2C:FD:A1:8D:80:BA
br0  2C:FD:A1:8D:80:B9 -> eth0 2C:FD:A1:8D:80:B9
eth0 2C:FD:A1:8D:80:B9 -> wan 2C:FD:A1:8D:80:B9
eth1 2C:FD:A1:8D:80:BB -> br-lan 2C:FD:A1:8D:80:BB
eth2 2C:FD:A1:8D:80:B9 -> lan 2C:FD:A1:8D:80:BB

Status:
Lan ports: Configured as oem port closer to power jack as wan and other as lan
Wlan: radio1@5Ghz 300 Mbps trouhgput. radio0@2.4Ghz 50Mbps troughput
Led: works
Buttons: reset works, wps untested.

Installation:
1. Reset device to orginal settings and enable ssh, set username and password: admin/password
2. Upload openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb
    `scp openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb admin@192.168.72.1:/tmp`
3. ssh to router:
   `ssh admin@192.168.72.1`
4. run in router:
    `mtd-write -d linux -i /tmp/openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb`
5. Reboot with command or with powercycling device
6. OpenWrt should boot and is ready after sort brust of flashing blue light
7. wget/scp openwrt-ipq40xx-generic-asus_map-ac1300-squashfs-sysupgrade.bin to /tmp dir of device
    `scp openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb root@192.168.1.1:/tmp`
8. ssh to OpenWrt device. 
    `ssh root@192.168.1.1 `
9.run command: 
   `sysupgrade -n /tmp/openwrt-ipq40xx-generic-asus_map-ac2200-squashfs-sysupgrade.bin`
10. After reboot OpenWrt is installed to device.

robimarko commented on target/linux/ipq40xx/base-files/etc/board.d/02_network:
```diff
@@ -65,9 +66,7 @@ ipq40xx_setup_interfaces()
-	cellc,rtl30vw)
-		ucidef_set_interface_lan "lan1 lan2"
-		;;
+	cellc,rtl30vw|\
```
I get what you are trying to do here, but it needs to be a separate commit

robimarko commented on target/linux/ipq40xx/base-files/lib/upgrade/platform.sh:
```diff
@@ -144,10 +145,8 @@ platform_do_upgrade() {
-		CI_KERNPART="linux"
-		nand_do_upgrade "$1"
-		;;
+	asus,map-ac2200 |\
```
Same here, it needs to be a separate commit

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-map-ac1300.dts:
```diff
@@ -0,0 +1,376 @@
+	};
+
+	soc {
+		rng@22000 {
```
Use existing labels for everything that has one, you should only be left with tcsr nodes here.

sbeach92 replied:
I'm not sure if I know how to do it. As this is copied from https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4019-map-ac2200.dts;h=ea39428b88c80d8760e75b7d63ec298b9e484cb4;hb=9ad3967f140bd6d08796f1ec151b753186b3eb44

Whole work is basically copy paste, but I try to find other supported devices that uses this led driver.

robimarko replied:
This has nothing to do with the LED driver but with the fact that I am constantly seeing just copy/paste that keeps on not using labels in the DTS for these nodes as there is no reason to keep manually overriding the nodes and using the existing label as for wifi etc.

So, instead of doing this hierarchy dance for `rng@22000` you can just use `&prng` in the same way as the UART nodes are.

sbeach92 replied:
Oh, so this goes one for every xxx@yyyyy should have altered with &zzz to make it right?

As you can probably see I'm not that experienced in programming, so appreciate your understanding.

robimarko replied:
Yes, every node that has labels in the SoC DTSI should use that.

Look at:
https://github.com/openwrt/openwrt/blob/main/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4029-ws-ap391x.dts

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-map-ac1300.dts:
```diff
@@ -0,0 +1,376 @@
+		   
+
+	flash@0 {
+
```
No need for this newline

hauke replied:
This was not done

robimarko commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-map-ac1300.dts:
```diff
@@ -0,0 +1,376 @@
+		reg = <1>;
+		spi-max-frequency = <24000000>;
+
+
```
Same here, extra newline

sbeach92 committed:
ipq40xx: add support for Asus Lyra Mini (MAP-AC1300)

Hardware
CHIPSET: IPQ4018 + QCA8072
LAN: 2x 1G
FLASH: GD5F1GQ4X 128 MiB + MX25L165D 2 MiB
RAM: 256 MiB
IO: Reset + WPS button
LED: Single Bright RGB led
POWER: 5V 2A

Flash layout similar to other asus brand routers.
Mac adresses extarcted from same mtd adress.
Created common .dtsi files as similar devices.

Orginal -> Openwrt
ath0 2C:FD:A1:8D:80:B8 -> phy0-ap0 2C:FD:A1:8D:80:B8
ath1 2C:FD:A1:8D:80:BA -> phy1-ap0 2C:FD:A1:8D:80:BA
br0  2C:FD:A1:8D:80:B9 -> br-lan   2C:FD:A1:8D:80:B9
eth0 2C:FD:A1:8D:80:B9 -> lan      2C:FD:A1:8D:80:B9
eth1 2C:FD:A1:8D:80:BB -> wan      2C:FD:A1:8D:80:BB

Status:
Lan ports: Configured as oem port closer to power jack as wan and other as lan
Wlan: radio1@5Ghz 300 Mbps trouhgput. radio0@2.4Ghz 50Mbps troughput
Led: works
Buttons: reset works, wps untested.

Installation:

1.  Reset device to orginal settings and enable ssh, set username and password: admin/password
2.  Upload openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb
    scp openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb admin@192.168.72.1:/tmp
3.  ssh to router:
    ssh admin@192.168.72.1
4.  run in router:
    mtd-write -d linux -i /tmp/openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb
5.  Reboot with command or with powercycling device
6.  OpenWrt should boot and is ready after sort brust of flashing blue light
7.  wget/scp openwrt-ipq40xx-generic-asus_map-ac1300-squashfs-sysupgrade.bin to /tmp dir of device
    scp openwrt-ipq40xx-generic-asus_map-ac1300-initramfs-uImage.itb root@192.168.1.1:/tmp
8.  ssh to OpenWrt device.
    ssh root@192.168.1.1
9.  run in router:
    sysupgrade -n /tmp/openwrt-ipq40xx-generic-asus_map-ac2200-squashfs-sysupgrade.bin
10. After reboot OpenWrt is installed to device.

Signed-off-by: Erkki Hietaranta <erkki.hietaranta@gmail.com>

sbeach92 commented:
> Please squash the changes

Im feeling that i didn't do this correctly, but not really sure how to avoid to have also commit merge branch 'openwrt:main'

raenye commented:
> > Please squash the changes
> 
> Im feeling that i didn't do this correctly, but not really sure how to avoid to have also commit merge branch 'openwrt:main'

Easiest is via `git rebase -i`. Just change all picks except the first to squash.

raenye commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-map-ac1300.dts:
```diff
@@ -0,0 +1,375 @@
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
```
Better call these `button-reset`, `button-eps`, etc.

raenye commented on target/linux/ipq40xx/image/generic.mk:
```diff
@@ -190,6 +190,20 @@ define Device/aruba_ap-365
+	DEVICE_VENDOR := ASUS
+	DEVICE_MODEL := Lyra Mini (MAP-AC1300)
+	SOC := qcom-ipq4018
+	DEVICE_PACKAGES := -kmod-ath10k-ct kmod-ath10k-ct-smallbuffers
```
256MB ram, so why small buffers?

raenye replied:
Another point here: the officially committed version should probably have the Candela-Tech (`-ct`) drivers, but if you want 802.11s (mesh) you need to self-compile with the non `-ct` version.

robimarko replied:
Please drop the wlan driver override from packages

raenye commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-map-ac1300.dts:
```diff
@@ -0,0 +1,375 @@
+	pinctrl-names = "default";
+	status = "okay";
+
+	led-controller@32 {
```
if this is taken from `qcom-ipq4019-map-ac2200.dts`, better move everything common to a dtsi file, say `qcom-ipq401x-map-acxx00.dtsi`, and include in both.

sbeach92 commented:
> > eth0 2C:FD:A1:8D:80:B9 -> wan 2C:FD:A1:8D:80:B9
> > eth1 2C:FD:A1:8D:80:BB -> br-lan 2C:FD:A1:8D:80:BB
> > eth2 2C:FD:A1:8D:80:B9 -> lan 2C:FD:A1:8D:80:BB
> 
> `eth0`,`eth2` were ports and `eth1` were the switch? `br-lan` is not a real device AFAIK.



> > eth0 2C:FD:A1:8D:80:B9 -> wan 2C:FD:A1:8D:80:B9
> > eth1 2C:FD:A1:8D:80:BB -> br-lan 2C:FD:A1:8D:80:BB
> > eth2 2C:FD:A1:8D:80:B9 -> lan 2C:FD:A1:8D:80:BB
> 
> `eth0`,`eth2` were ports and `eth1` were the switch? `br-lan` is not a real device AFAIK.

I returnet to stock software and it seems i have swapped wan adresses of wan and lan.
I belive eth0 and eth1 were ports and correct counterpart to br-lan is br0.

Original ifconfig -a
https://pastebin.com/0Ld69smb

Openwrt ifconfig -a where lan and wan mac corrected
https://pastebin.com/myZRL2Pm

sbeach92 commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq401x-map-acxx00.dtsi:
```diff
@@ -0,0 +1,171 @@
+#include <dt-bindings/leds/common.h>
+
+/ {	
+	aliases {
```
I was assuming that updating map ac2200 would need separate pull request

robimarko replied:
If they share stuff commonly then make a commit that splits out the original device into DTSI+DTS and then just leverage that in new device commit.

sbeach92 replied:
As they share things, Map-ac2200 needs same corrections to its dts as this pullreguest have.

So can I split original device, make corrections and still use pullreguest ipq40xx: Add support for Asus Lyra Mini but with seperate commit?

hauke commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq401x-map-acxx00.dtsi:
Do you plan to add more similar devices?
If not I would prefer if you add the changes from the dtsi file into your dts file and only have one dts file.

raenye replied:
@hauke there are AC-2200 (already supported) and AC-1300 (the device OP added support for), and they have some commonalities.

hauke replied:
If it makes sense you should then change them to use the common dtsi file. Please extract the common parts in a separate commit before the one adding the new device.

raenye commented on target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -54,6 +54,7 @@ case "$FIRMWARE" in
 "ath10k/pre-cal-ahb-a000000.wifi.bin")
 	case "$board" in
 	asus,map-ac2200|\
+	asus,map-ac1300|\
```
maintain alphabetic order please.

raenye commented on target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -151,6 +152,7 @@ case "$FIRMWARE" in
 "ath10k/pre-cal-ahb-a800000.wifi.bin")
 	case "$board" in
 	asus,map-ac2200|\
+	asus,map-ac1300|\
```
maintain alphabetic order please.

raenye commented on target/linux/ipq40xx/base-files/lib/upgrade/platform.sh:
```diff
@@ -6,7 +6,8 @@ RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
 
 platform_check_image() {
 	case "$(board_name)" in
-	asus,rt-ac42u |\
```
there shouldn't have been any change in the rt-ac42u line IMHO.

sbeach92 replied:
As there hasn't been real change, I have to figure out how to get git to see that line is there as in orginal. Learning curve is steep. Manual diff edit or something?

raenye replied:
Maybe it's tabs vs spaces?

sbeach92 committed:
ipq40xx: Split map-ac2200 and rt-ac42u .dts to common .dtsi

Split qcom-ipq4019-rt-ac42u.dts to qcom-ipq4019-asus.dtsi
Splir qcom-ipq4019-map-ac2200.dts to qcom-ipq4019-asus.dtsi and qcom-ipq401x-map-acxx00.dtsi

Converted manually overridden labels to ones that exsists in qcom-ipq2019.dtsi

Signed-off-by: Erkki Hietaranta <erkki.hietaranta@gmail.com>

raenye commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rt-ac42u.dts:
```diff
@@ -207,59 +127,23 @@
+			#trigger-source-cells = <0>;
 		};
-	};
+	};		
```
extra space here.

raenye commented:
I think you need to `git rebase -i` and reorder the commits -- AC1300 should be the last commit.

raenye commented:
Commits still seem to be in reverse order. The first ones to be applied should do the splitting into dtsi, and the last one is the one that should introduce MAP-AC1300.

raenye commented:
Rebase -i is your friend.
Drop all merges.

sbeach92 commented:
I really can't figure out how to reopen this pullreguest as my pull -f results does not showup on this pullreguest.

robimarko commented:
You need to force push your changes and then just click reopen in GH web ui

sbeach92 commented:
> You need to force push your changes and then just click reopen in GH web ui

I tried many times but that seemed not to be a option to do even with these instructions.
https://openwrt.org/docs/guide-developer/working-with-github-pr#reopen_closed_pr

But now i

robimarko commented:
Oh, this is something new for me, did not know that they require no changes in order to reopen.

raenye commented:
> No new commits in \"ASUS-LYRA-MINI-MAP1300\

That seems to be the problem - there are no commits in the PR. I think you need to have some changes in the forced-push branch and then reopen.

DragonBluep commented:
Please try:
`git push -f origin 08e8e181dc4cff48ad6a59260ee7d7957849a3d3:ASUS-LYRA-MINI-MAP1300`

sbeach92 commented:
> Please try: `git push -f origin 08e8e181dc4cff48ad6a59260ee7d7957849a3d3:ASUS-LYRA-MINI-MAP1300`

git push -f origin 08e8e181dc4cff48ad6a59260ee7d7957849a3d3:ASUS-LYRA-MINI-MAP1300
Enumerating objects: 3846, done.
Counting objects: 100% (3846/3846), done.
Delta compression using up to 2 threads
Compressing objects: 100% (997/997), done.
Writing objects: 100% (2163/2163), 963.66 KiB | 15.30 MiB/s, done.
Total 2163 (delta 1571), reused 1525 (delta 1054)
remote: Resolving deltas: 100% (1571/1571), completed with 813 local objects.
To https://github.com/sbeach92/openwrt.git
 + c9e804d1ca...08e8e181dc 08e8e181dc4cff48ad6a59260ee7d7957849a3d3 -> ASUS-LYRA-MINI-MAP1300 (forced update)


I have tried this many times, but seems that somehow push -f wont show up in this PR, even PR links back to my branch, there is no link back to this PR. I leave my branch to this state, but im not holding my breath that this will solve this problem.

sbeach92 commented:
Now i again have button to do reopen, but will it work?

DragonBluep commented:
> Now i again have button to do reopen, but will it work?
> 
> Edit: Reopen and comment just added comment

Try to add another new commit and push it again.
```
echo test > ./test
git add -f ./test
git commit -m test
git push
```

sbeach92 commented:
> > Now i again have button to do reopen, but will it work?
> > Edit: Reopen and comment just added comment
> 
> Try to add another new commit and push it again.
> 
> ```
> git checkout ASUS-LYRA-MINI-MAP1300 
> git reset --hard 08e8e181dc4cff48ad6a59260ee7d7957849a3d3
> echo test > ./test
> git add -f ./test
> git commit -m test
> git push
> ```

```
user@distro:~/openwrt_asus/openwrt$ git checkout ASUS-LYRA-MINI-MAP1300 
Already on 'ASUS-LYRA-MINI-MAP1300'
Your branch and 'origin/ASUS-LYRA-MINI-MAP1300' have diverged,
and have 7 and 1 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
user@distro:~/openwrt_asus/openwrt$ git reset --hard 08e8e181dc4cff48ad6a59260ee7d7957849a3d3
HEAD is now at 08e8e181dc ipq40xx: Split map-ac2200 and rt-ac42u .dts to common .dtsi
user@distro:~/openwrt_asus/openwrt$ echo test > ./test
user@distro:~/openwrt_asus/openwrt$ git add -f ./test
user@distro:~/openwrt_asus/openwrt$ git commit -m test
[ASUS-LYRA-MINI-MAP1300 319bd4cc81] test
 1 file changed, 1 insertion(+)
 create mode 100644 test
user@distro:~/openwrt_asus/openwrt$ git push
Enumerating objects: 4, done.
Counting objects: 100% (4/4), done.
Delta compression using up to 2 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 275 bytes | 11.00 KiB/s, done.
Total 3 (delta 1), reused 1 (delta 0)
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/sbeach92/openwrt.git
   08e8e181dc..319bd4cc81  ASUS-LYRA-MINI-MAP1300 -> ASUS-LYRA-MINI-MAP1300
```

sbeach92 committed:
ipq40xx: Split ipq4018-rt-ac58u.dts to common .dtsi

Split qcom-ipq4018-rt-ac58u.dts to qcom-ipq4018-asus.dtsi and qcom-ipq-rt-ac58u.dts
Converted manualy overridden labels to ones that exsist in qcom-ipq4019.dtsi

Signed-off-by: Erkki Hietaranta <erkki.hietaranta@gmail.com>

sbeach92 commented:
> As they share things, Map-ac2200 needs same corrections to its dts as this pullreguest have.
> 
> So can I split original device, make corrections and still use pullreguest ipq40xx: Add support for Asus Lyra Mini but with seperate commit?
>>
>> If they share stuff commonly then make a commit that splits out the original device into DTSI+DTS and then just leverage that in new device commit.
>>>
>>> As they share things, Map-ac2200 needs same corrections to its dts as this pullreguest have.
>>> 
>>> So can I split original device, make corrections and still use pullreguest ipq40xx: Add support for Asus Lyra Mini but with seperate commit?

>> Great, now you can squash everything

So i squash commits that splits DTS and corrections that are made to MAP-AC2200 ?

robimarko commented:
Wait, I took another look.

Why are MAP-AC2200 changes in this PR?

sbeach92 commented:
> Wait, I took another look.
> 
> Why are MAP-AC2200 changes in this PR?

1.  I'm not sure if they need separete PR:s and you answer tells about separate commits. I tried to ask clarification
2.  As i have MAP-AC2200 device in hand i made corrections which could be there because of split.
3.  I was thinking that if more than one commit per PR is possible i could do ther fixes allso.
4. Point 3. makes sens to me as i learned that you cant have deppend PRs

sbeach92 commented:
> Could be a useful comment for you too: [#13877 (comment)](https://github.com/openwrt/openwrt/pull/13877#issuecomment-1843923308)

And eventually this was the fix, atleast renaming branch

robimarko commented:
> 1. I'm not sure if they need separete PR:s and you answer tells about separate commits. I tried to ask clarification
> 2. As i have MAP-AC2200 device in hand i made corrections which could be there because of split.
> 3. I was thinking that if more than one commit per PR is possible i could do other fixes allso.
> 4. Point 3. makes sens to me as i learned that you cant have deppend PRs

Of course that more than 1 commit is allowed in a PR, that's the whole point of a PR.
But you want to do one thing as the focus of the PR.

Let me ask one question, is MAP-AC2200 working fine now or are those fixes required?

sbeach92 commented:
> > 1. I'm not sure if they need separete PR:s and you answer tells about separate commits. I tried to ask clarification
> > 2. As i have MAP-AC2200 device in hand i made corrections which could be there because of split.
> > 3. I was thinking that if more than one commit per PR is possible i could do other fixes allso.
> > 4. Point 3. makes sens to me as i learned that you cant have deppend PRs
> 
> Of course that more than 1 commit is allowed in a PR, that's the whole point of a PR. But you want to do one thing as the focus of the PR.
> 
> Let me ask one question, is MAP-AC2200 working fine now or are those fixes required?

It is working fine, exept maybe missing BDF for QCA9888, allso seemsto be decsribed here: https://github.com/openwrt/openwrt/pull/13604

sbeach92 commented on target/linux/ipq40xx/base-files/lib/preinit/05_set_iface_mac_ipq40xx.sh:
```diff
@@ -2,10 +2,17 @@
 preinit_set_mac_address() {
 	case $(board_name) in
+	asus,map-ac1300)
+		base_mac=$(mtd_get_mac_binary_ubi Factory 0x1006)
+		ip link set dev eth0 address $(macaddr_add "$base_mac" 1)
+		ip link set dev lan address $(macaddr_add "$base_mac" 1)
+		ip link set dev wan address $(macaddr_add "$base_mac" 3)
+		;;		
 	asus,map-ac2200)
 		base_mac=$(mtd_get_mac_binary_ubi Factory 0x1006)
 		ip link set dev eth0 address $(macaddr_add "$base_mac" 1)
-		ip link set dev eth1 address $(macaddr_add "$base_mac" 3)
+		ip link set dev wan address $(macaddr_add "$base_mac" 1)
+		ip link set dev lan address $(macaddr_add "$base_mac" 3)
```
Wan and Lan macs are reversed.

I kept Map-ac2200 fixes in separate commit and I'm planning to do separate PR for those fixes. But If I go down that path, I think I have to wait until this PR is merged

sbeach92 commented on target/linux/ipq40xx/base-files/etc/board.d/02_network:
```diff
@@ -161,6 +162,16 @@ ipq40xx_setup_macs()
 	8dev,habanero-dvk)
 		label_mac=$(mtd_get_mac_binary "ART" 0x1006)
 		;;
+	asus,map-ac1300)
+		label_mac=$(mtd_get_mac_binary_ubi Factory 0x1006)
+		lan_mac=$(macaddr_add "$label_mac" 1)
+		wan_mac=$(macaddr_add "$label_mac" 3)
+		;;
+	asus,map-ac2200)
+		label_mac=$(mtd_get_mac_binary_ubi Factory 0x1006)
+		wan_mac=$(macaddr_add "$label_mac" 1)
+		lan_mac=$(macaddr_add "$label_mac" 3)
+		;;
```
Wan & Lan reversed here allso

sbeach92 commented on target/linux/ipq40xx/image/generic.mk:
```diff
@@ -190,12 +190,24 @@ define Device/aruba_ap-365
 	DEVICE_VENDOR := ASUS
 	DEVICE_MODEL := Lyra (MAP-AC2200)
 	SOC := qcom-ipq4019
-	DEVICE_PACKAGES := ath10k-firmware-qca9888-ct kmod-ath3k
```
as for map ac1300 the comment was to remove driver override made it here too

https://github.com/openwrt/openwrt/pull/13571#discussion_r1347715879

They weren't there. I will add them. Should JFFS2 to be removed if Image_size is defined?

sbeach92 replied:
Will be resolved in PR after this get merged

robimarko commented:
Can you please rebase this, I will give it another review and merge if ready

robimarko commented:
Now that there is NVMEM on UBI support, please use that instead of scripts for MAC setting

Djfe commented:
@robimarko this device was just merged, could you close this older pr please? :)
